Javadocs version 1 complete

albydeca opened the pull request:
Hi, I have finished creating the first version of the Java SDK documentation on the Wiki. As suggested this closely follows the NodeJS docs.

One issue I would flag is that I am unable to see syntax highlighting for the many Java snippets I have included, despite this highlighting working perfectly in VSCode. Any ideas why this is?

Please let me know what you think. I'm tagging @mastrogiovanni as he is my goto developer at the Foundation.

Cheers :)

mastrogiovanni commented:
Hi @albydeca can you update the documentation so that conflicting files are solved?

aleksei-korolev commented on documentation/docs/basics/authentication/example_java.md:
```diff
@@ -0,0 +1,211 @@
+
+##  Putting It All Together
+
+You can find the complete code example [here](https://github.com/albydeca/iota-is-sdk/blob/main/src/main/java/net/gradbase/clients/BaseClient.java). All snippets above are taken from there.
```
@mastrogiovanni where should we store the example source code?

aleksei-korolev commented on documentation/docs/examples/java/create-channel.md:
```diff
@@ -0,0 +1,67 @@
+# Create Channel
+
+The [example-5](https://github.com/albydeca/iota-is-sdk/blob/main/examples/src/main/java/net/gradbase/examples/CreateChannel.java)
+script authenticates an Integration Service client to manage Identities using the Admin identity created in [example-0](how-to-run-examples) and then performs the following tasks:
```
```suggestion
script authenticates an Integration Services client to manage identities using the admin identity created in [example-0](how-to-run-examples) and then performs the following tasks:
```

identity and admin identity: these are just technical terms, no need to capitalize technical terms. An Integration Services client: a client is a technical term, so non-capitalized, Integration Services is a name, so capitalized. Same across the rest of the contribution. No worries, this is a confusing rule.

aleksei-korolev commented on documentation/docs/examples/java/create-identity-and-credentials.mdx:
```diff
@@ -0,0 +1,85 @@
+
+## Authentication
+
+You can use the following script to authenticate the client using the Admin identity and get the
+BasicIdentityCredential that the admin owns: this VC (verifiable credential),
+which is required for an Issuer to sign verifiable credentials for other Identities.
```
```suggestion
You can use the following script to authenticate the client using the admin identity and get the
BasicIdentityCredential that the admin owns: this is a verifiable credential
which is required for an issuer to sign verifiable credentials for other Identities.
```

aleksei-korolev commented on documentation/docs/examples/java/delete-users.md:
````diff
@@ -0,0 +1,48 @@
+```
+
+:::note
+You can only remove an identity in the Integration Services backend, **the Identity cannot be removed from the Tangle**.
````
```suggestion
You cannot delete an identity from the Tangle. You can only remove it from the Integration Services database.
```

You can emphasise important information by simply putting it first.

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
```diff
@@ -0,0 +1,92 @@
+
+* `IdentityClient` allows the developer to create, update, delete users, add/revoke/check their credentials and manage trusted authorities. Self-Sovereign Identity (SSI) developers should make use of this part.
+* `ChannelClient` allows the developer to create, delete data channels (IOTA Streams), manage all subscription-related actions for those channels, push and pull data. IoT developers might want to use this to implement M2M communications.
+* `examples` provides a set of sample activities that show the potential of what can be done with the codebase, and also acts as a rudimentary testsuite. More about these can be found in the [dedicated section](../../../examples/java/introduction.mdx).
```
```suggestion
* `examples` provides a set of sample activities that show the potential of what can be done with the codebase, and also acts as a rudimentary test suite. More about these can be found in the [dedicated section](../../../examples/java/introduction.mdx).
```

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
````diff
@@ -0,0 +1,92 @@
+The JAR can be used as a dependency to run the examples, which, contrary to the node implementation, exist as part of their own package
+which depends on this project's JAR (see `examples/pom.xml` in the repo)
+
+Note that the sensitive variables are accessed in the code as follows:
+```java
````
```suggestion
Note that the sensitive variables are accessed in the code as follows:

```java
```

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
````diff
@@ -0,0 +1,92 @@
+```bash
+mvn clean install
+```
+You are now ready to use the JAR and access the classes. Please remember to keep the `env.properties` *in the same folder as the JAR*.
````
```suggestion

You are now ready to use the JAR and access the classes. Please remember to keep the `env.properties` *in the same folder as the JAR*.
```

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
```diff
@@ -0,0 +1,92 @@
+
+The codebase is divided into three parts:
+
+* `IdentityClient` allows the developer to create, update, delete users, add/revoke/check their credentials and manage trusted authorities. Self-Sovereign Identity (SSI) developers should make use of this part.
```
```suggestion
* `IdentityClient` allows the developer to create, update, delete users and add, revoke, and check their credentials, and manage trusted authorities. Self-Sovereign Identity (SSI) developers should make use of this part.
```

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
````diff
@@ -0,0 +1,92 @@
+    this.jwt = null;
+}
+```
+Anyone who wishes to implement such access in a more convenient way for their needs is welcome to make the changes and submit a pull request.
````
```suggestion

Anyone who wishes to implement such access in a more convenient way for their needs is welcome to make the changes and submit a pull request.
```

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
````diff
@@ -0,0 +1,92 @@
+
+Please set up the following files in order to generate the JAR and run the code locally:
+
+- `env.properties` - with the following structure:
+  ```
+  api-key=XXXXXXX
+  api-version=vX.X
+  api-url=XXXXXXX
+  identity-file=adminIdentity.json
+- `adminIdentity.json` - will contain the admin identity object (json file with elements `doc` and `key`)
+
+Then, perform the following steps:
+```bash
+mvn clean install
+```
````
I think you are missing triple ticks here somewhere.

albydeca replied:
indeed :)

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
```diff
@@ -0,0 +1,92 @@
+
+or download JAR from [MVNRepository](https://mvnrepository.com/artifact/net.gradbase/iota.is.sdk/0.0.1)
+
+or simply add to your POM:
```
```suggestion
or simply add to your POM:

```

aleksei-korolev commented on documentation/docs/getting_started/installation/java/local_setup.md:
```diff
@@ -0,0 +1,92 @@
+
+## Download and Build the Project
+
+Build from source
```
```suggestion
Build from source:

```

aleksei-korolev commented on documentation/docs/examples/java/introduction.mdx:
````diff
@@ -0,0 +1,136 @@
+If you have a JSON Identity, you can authorize your client as follows:
+
+```java
+  client.authenticate(newUserIdentity.getJSONObject("doc").getString("id"),
+			  newUserIdentity.getJSONObject("key").getString("public"),
+				newUserIdentity.getJSONObject("key").getString("secret"));
````
Would it make sense to introduce local variables to make the example more readable?

aleksei-korolev commented on documentation/docs/examples/java/introduction.mdx:
```diff
@@ -0,0 +1,136 @@
+  System.out.println("username: " + randomUsername);
+  JSONObject newUserIdentity = client.create(randomUsername, claim);
+  System.out.println("created new user " + newUserIdentity);
+
```
```suggestion
```

aleksei-korolev commented on documentation/docs/examples/java/introduction.mdx:
```diff
@@ -0,0 +1,136 @@
+
+## Install
+
+Build from source
```
```suggestion
Build from source:

```

aleksei-korolev reviewed (changes requested):
Hi @albydeca, I am one of the technical writers at IF. Thank you for your contribution, I have just finished reviewing it. Please don't be discouraged by the amount of change requests, these are mostly minor formatting and styling fixes. All in all, your contribution is solid.

Cheers

albydeca commented on documentation/docs/basics/authentication/example_java.md:
````diff
@@ -0,0 +1,211 @@
+The last step is to request the JWT. You can use the following code snippet with your signed nonce in the request body.
+
+```java
+    [...]
+        final String endpoint = "authentication/prove-ownership/" + didId;
````
yes. Sorry about that

albydeca commented:
Hi all,

Thanks to @aleksei-korolev for the useful suggestions! I have now implemented the changes.

As for the conflicts, most of them stem from the fact that I have changed the directory structure in the `examples/` folder, in particular I created two subfolders: `node/` and `java/`. The rest of the conflicts are in files with minimum intervention on my part. It may be that my changes are conflicting with newer ones on the master branch.

It might make sense for us to get on a call to better understand how we can deal with this. Apart from the changes in directory structure, the rest should be immediate to solve.
